gr-fec/lib/cc_decoder_impl.cc

```diff
@@ -45,36 +47,34 @@ cc_decoder_impl::cc_decoder_impl(int frame_size,
                                  cc_mode_t mode,
                                  bool padded)
     : generic_decoder("cc_decoder"),
-      d_k(k),
-      d_rate(rate),
+      d_parity_table(initial_parity_table()),
+      d_max_frame_size(frame_size),
+      d_frame_size(frame_size),
       d_polys(polys),
       d_mode(mode),
-      d_padding(0),
+      d_padding(
+          /* set up a padding factor. If padding, the encoded frame was extended by this
+           * many bits to fit into a full byte.
+           */
+          padded && mode == CC_TERMINATED
+              ? static_cast<int>(8.0f * ceilf(s_rate * (s_k - 1) / 8.0f) -
+                                 (s_rate * (s_k - 1)))
+              : 0),
       d_start_state_chaining(start_state),
       d_start_state_nonchaining(start_state),
       d_end_state_nonchaining(end_state)
 {
-    // Set max frame size here; all buffers and settings will be
-    // based on this value.
-    d_max_frame_size = frame_size;
-    d_frame_size = frame_size;
-
-    // set up a padding factor. If padding, the encoded frame was extended
-    // by this many bits to fit into a full byte.
-    if (padded && (mode == CC_TERMINATED)) {
-        d_padding = static_cast<int>(8.0f * ceilf(d_rate * (d_k - 1) / 8.0f) -
-                                     (d_rate * (d_k - 1)));
+    if (k != 7 || rate != 2) {
+        // used to throw std::runtime_error("cc_decoder: parameters not supported");
+        throw std::invalid_argument(
+            "cc_decoder: Only k=2, rate=2 convolutional codes are supported");
```

Comment:
I know this has already been merged, but I was reading through the changes and saw a minor typo. This should read `k=7`.